sdk/containerregistry/container-registry/src/utils/digest.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import crypto from "crypto";
+
+export function calculateDigest(buffer: Buffer): Promise<string>;
+
+export function calculateDigest(stream: NodeJS.ReadableStream): Promise<string>;
+
+export function calculateDigest(bufferOrStream: NodeJS.ReadableStream | Buffer): Promise<string> {
+  const hash = crypto.createHash("sha256");
+  if (Buffer.isBuffer(bufferOrStream)) {
+    return Promise.resolve(`sha256:${hash.update(bufferOrStream).digest("hex")}`);
+  } else {
+    bufferOrStream.pipe(hash);
+    return new Promise((resolve) =>
+      bufferOrStream.on("end", () => {
```

Comment:
this doesn't do any error handling on the stream, should we think about rejecting the promise on error events?